### ISSUE TITLE
[MQ] color-gamut Keywords

### ISSUE THREAD

**Issue by Crissov** [w3c/csswg-drafts#4535]
The Color Display Quality  [`color-gamut`](https://drafts.csswg.org/mediaqueries/#color-gamut) media-query currently supports three keyword values:

* `srgb`
* `p3`
* `rec2020`

Alas, they explicitly (though perhaps not intentionally) do not exactly match the similar colorspace keywords in [css-color](https://drafts.csswg.org/css-color/#predefined) in neither keyword nor meaning: 

* `srgb` (identical)
* `display-p3` (DCI uses a different reference illuminant optimized for cinemas, rather than the common D65)
* `rec-2020` (note the hyphen!)
* (`a98-rgb`)
* (`prophoto-rgb`)

It seems that more descriptive terms would be more appropriate, though Iʼm not exactly sure which:

- expanse: `narrow` / `limited`, `normal`, `wide`, `wider` / `extra-wide`, `natural`
- target technology: `crt`, `lcd`, `led`, `oled`, `laser`, `ink`, … 
- primary medium: `screen`, `cinema`, `print`, `photo`, … 
- television definition:
  - `sd`, `hd`, `uhd`
  - `sdtv`, `hdtv`, `uhdtv`; `hdr`
- standard:
  - `rec-470`/`rec-601`, `rec-709`, `rec-2020`; `rec-2100`; … 
  - `itu-r-bt470`/`itu-r-bt601`, `itu-r-bt709`, `itu-r-bt2020`, `iso-22028-2`, `iso-22028-3`, `iso-22028-4`, `iec-61966-2-1`, `iec-61966-2-2`, `smpte-eg-432-1`, `smpte-rp-431-2`
  - `bt-470`/`bt-601`, `bt-709`, `bt-2020`; `bt-2100`; `iso-2`, `iso-3`, `iso-4`; `iec-1`/`iec-2-1`, `iec-2`/`iec-2-2`; `eg-432`/`smpte-1`, `rp-431`/`smpte-2`
- …

**Comment by AmeliaBR** [comment]
Relevant discussion at the end of the thread about color-profile names in #4056 

Also note that Chromium and WebKit have shipped [support for the color-gamut MQ](https://developer.mozilla.org/en-US/docs/Web/CSS/@media/color-gamut#Browser_compatibility). Doesn't mean we can't change to using keywords that are less likely to be misinterpreted, but the existing keywords might need to be kept around for compat.

**Comment by tabatkins** [comment]
Yeah, the differences are unintentional; we recently changed the color() keywords to have more reasonable hyphenation, but didn't think to go update the similar ones in color-gamut.

As Amelia says, the current keywords are already widely shipping, and probably shouldn't be changed. But I think it's cheap to add more keywords when they're just synonyms.

I agree that, unless there's a significant objection, we should make sure that the color() keywords and the corresponding color-gamut keywords should match exactly.

**Comment by svgeesus** [comment]
@tabatkins I suggest not, because they mean different things [as I explained](https://github.com/w3c/csswg-drafts/issues/4056#issuecomment-557838346)

**Comment by Crissov** [comment]
… and thatʼs why I think that very different keywords should be used, even if the current, similar ones would need to be kept for compatibility with existing implementations.

**Comment by svgeesus** [comment]
`technology` and `medium` look very like the old Media Types that were found unworkable and abandoned.

If we were designing this from scratch and no-one had shipped, then `normal` | `wide` | `ultrawide` or something like that would be reasonable (but then, people would complain and ask for more specific examples of what exactly is wide, for example is P3 wide or ultrawide, etc).

Given that this has shipped, I am reluctant to add aliases which would be partly supported for some time. But I won't oppose it if the CSS WG decides to add aliases.

**Comment by svgeesus** [comment]
I'm less than enthusiastic about the suggestion on the Dec 11 call to roll back the [consistent hyphenaton](https://github.com/w3c/csswg-drafts/issues/4056#issuecomment-546539874) for built-in color profiles.

I agree with @AmeliaBR  and @fantasai that is we make them exactly the same as the built-in colorspaces, then people will assume they are testing for *that exact profile* and will then also want a bunch of other profiles added. This is very different to the 'roughly sRGB' | `common wide gamut` | `superwide 4k TV gamut` buckets for MQ.

As I said earlier, if this was being designed now then maybe better keywords could be used but then there would be a drawback on that. Given that it shipped already in 2 browsers the cost of bikeshedding (there is always a cost, but this time it is high) seems excessive.

**Comment by tabatkins** [comment]
I'm still strongly on the position that "if they're exactly the same people will expect them to be the same" is a post hoc rationalization, and "these are spelled slightly differently, now nobody will make that assumption" is simply false. 

What'll actually happen is people will be confused and annoyed by the seeming lack of consistency, and have trouble learning and remembering which is which.

The two things are using these keywords to mean approximately the same things. We should just be consistent in the naming.  I agree it's not worth changing the `(color-gamut)` keywords, which means I'm still strongly supportive of switching the `color()` keywords back to being consistent.

**Comment by svgeesus** [comment]
(This change also rolls-back the hyphenation introduced in https://github.com/w3c/csswg-drafts/issues/4056 for rec-2020 only).

**Comment by Crissov** [comment]
Shouldn't this media query also include a value for devices with color gamuts significantly narrower than sRGB? Wouldn't that be reason enough to add (recommended) aliases for all of them?

**Comment by tabatkins** [comment]
CSS has not traditionally recognized the existence of narrower-than-sRGB gamuts, so no.

**Comment by Crissov** [comment]
The gamut of CMYK devices is not necessarily narrower than sRGB, but certainly shaped differently. Iʼm not sure whether that matters.

**Comment by jstblck** [comment]
I just noticed this thread. I'm the person who proposed the hyphens in the `color` keywords. I see this is resolved but I just want to add the following for the record.

First, (for what it's worth) I'm completely fine that `rec-2020` has been changed (back) to `rec2020`.

Second, I think `rec2020` is still compatible with the hyphenation scheme proposed [here](https://github.com/w3c/csswg-drafts/issues/4056#issuecomment-511172542). It simply depends on whether the period in "Rec.2020" is considered a space or not.

Therefore, in my opinion, `rec-2020` and `rec2020` are both compatible with the naming scheme for css-color and I'm happy that `rec2020` is being used for consistency between `color` and `color-gamut` as well as compatibility with the shipping browsers.
